### PR TITLE
web: add pinned resources to the overview grid

### PR DESCRIPTION
**Pull request by nicks**
Hello @hyu, @landism,

Please review the following commits I made in branch nicks/ch10621:

4f3fe464d251686b0b422281da57b8419a2f74f5 (2021-01-07 10:59:49 -0500)
web: add pinned resources to the overview grid

15a248b16ebc64b8f92f3aa859f146868889d69b (2021-01-06 18:05:08 -0500)
web: refactor sidebar context now that pins will appear in more places
add an in-memory implementation for storybook

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

**Comment by nicks** [comment]
Screenshot from 2021-01-07 10-55-53

File: web/src/OverviewPane.tsx [hunk 1]
```diff
@@ -19,48 +22,63 @@ let OverviewPaneRoot = styled.div`
...
-let AllServicesLabel = styled.div`
-  font: ${Font.sansSerif};
-  color: ${Color.blue};
-  margin: 12px;
-`
+function PinnedServicesDivider() {
+  return (
+    <ServicesDividerRoot>
+      <GridDividerPinSvg style={{ marginLeft: "28px" }} />
+      <ServicesLabel>Favorite Resources</ServicesLabel>
```
Comment (hyu) [inline web/src/OverviewPane.tsx:47]: Let's consistently refer to these as "Pinned," like in the sidebar. "Fave" would be if we dared to use the cute lil 💕 metaphor...

**Review by hyu — APPROVED** [review]
Nice! Now we just need to nail down the actual UI for pinning items in the Overview...

File: web/src/OverviewPane.tsx (hunk 1 above)
Comment (nicks) [reply web/src/OverviewPane.tsx]: done!

**Pull request merged**